leakcanary-object-watcher-android/src/main/java/leakcanary/internal/DefaultCanaryLog.kt

```diff
@@ -9,17 +9,12 @@ internal class DefaultCanaryLog : Logger {
     message: String,
     vararg args: Any?
   ) {
-    val formatted = if (args.isNotEmpty()) {
-      String.format(message, *args)
-    } else {
-      message
-    }
+    val formatted = if (args.isNotEmpty()) String.format(message, args) else message
+
     if (formatted.length < 4000) {
       Log.d("LeakCanary", formatted)
     } else {
-      val lines = formatted.split("\n".toRegex())
```

Comment:
Regular expression was getting compiled every time, so I extracted it to companion object. Also removed redundant conversion to typed array

Reply:
wow, nice catch!